shuup/admin/modules/contacts/object_selector.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shuup Commerce Inc. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.db.models import Q
+from typing import Iterable, Tuple
+
+from shuup.admin.utils.permissions import has_permission
+from shuup.admin.views.select import BaseAdminObjectSelector
+from shuup.core.models import CompanyContact, Contact, PersonContact
+
+
+class ContactAdminObjectSelector(BaseAdminObjectSelector):
+    ordering = 3
+
+    @classmethod
+    def handles_selector(cls, selector):
+        return selector == "shuup.%s" % Contact._meta.model_name
+
+    def has_permission(self, user):
+        return has_permission(user, "contact.object_selector")
+
+    def get_objects(self, search_term, *args, **kwargs) -> Iterable[Tuple[int, str]]:
+        """
+        Returns an iterable of tuples of (id, text)
+        """
+        shop = kwargs.get("shop")
```

Comment:
same here, do not let the user to select any shop, should only allow the user to filter the shops the user has access to.